### PR TITLE
Primary Detail: use new raised card variant in demo, add bulk select

### LINKED ISSUE #7312: Primary-detail demo - update to use new selectable card
Want to update this demo to utilize the new selectable card styling: https://www.patternfly.org/v4/demos/primary-detail/react-demos/primary-detail-card-view-expanded/

Core PR: https://github.com/patternfly/patternfly/pull/4810

### PR DESCRIPTION
**Pull request by jenny-s51**
**What**: Closes #7312 

Added some work here that applies towards #7657 - this demo now supports bulk select, and the ability to delete cards. 

However in order to fully align this demo with core, a lot of additional logic/functionality is required, which is out of scope of the original issue to use the new raised variant for the card. For that reason I opened #7657

**Additional issues**: #7657

### CODE REVIEW

File: packages/react-core/src/demos/PrimaryDetail.md [hunk 1]
```diff
@@ -1140,6 +1170,107 @@ class PrimaryDetailCardView extends React.Component {
...
+      });
+    };
+
+    this.onCardKebabDropdownSelect = (key, _event) => {
+      this.setState({
+        [key]: !this.state[key]
+      });
+    };
+
+    this.deleteItem = item => _event => {
```
[the remark this answers is not in the document]
Comment (jenny-s51) [reply packages/react-core/src/demos/PrimaryDetail.md:1259]: Nice catch @nicolethoen - thank you! Updated.

**Review by nicolethoen — APPROVED** [review]
LGTM!

**Comment by nicolethoen** [comment]
@mmenestr this is the direct URL to the updated demo: https://patternfly-react-pr-7658.surge.sh/demos/primary-detail/react-demos/primary-detail-card-view/

**Review by thatblindgeye — COMMENTED** [review]
Overall looks good! A couple of things I'm noticing (which these may be better as followup issues) are:

- When clicking anywhere in the card, the drawer gets toggled open. This includes clicking the checkbox or kebab toggle. The expectation I'd have is clicking either of those would only trigger those specific elements, and clicking anywhere else in the card would select it.
- This I can create a spinoff issue for since it might require tweaking the logic of selectable cards a bit, but should we have the card drawers closable in ways other than the "close" button? When navigating via keyboard, once I click a card to open the drawer it remains open and blocks other cards completely, and you'd have to tab through every single card to get to the close button.

@jenny-s51 what do you think?

**Review by jenny-s51 — COMMENTED** [review]
Thank you for your awesome feedback @thatblindgeye !! 

> * When clicking anywhere in the card, the drawer gets toggled open. This includes clicking the checkbox or kebab toggle. The expectation I'd have is clicking either of those would only trigger those specific elements, and clicking anywhere else in the card would select it.

Definitely makes sense to have the checkbox, drawer toggle, and kebab toggles all behave independently! I've updated the code to separate all the different toggling behaviors you listed here. 

> * This I can create a spinoff issue for since it might require tweaking the logic of selectable cards a bit, but should we have the card drawers closable in ways other than the "close" button? When navigating via keyboard, once I click a card to open the drawer it remains open and blocks other cards completely, and you'd have to tab through every single card to get to the close button.

Nice catch!! I fixed this by making sure the drawer closes if a user selects the card a second time either via mouse or keyboard. Seems like a good, working solution for this! WDYT @thatblindgeye ?

File: packages/react-core/src/demos/PrimaryDetail.md [hunk 2]
```diff
@@ -1095,22 +1101,26 @@ class PrimaryDetailCardView extends React.Component {
...
+      );
     };
 
     this.onCardClick = event => {
       if (event.currentTarget.id === this.state.activeCard) {
+        this.setState({
+          isDrawerExpanded: !this.state.isDrawerExpanded
+        });
         return;
       }
```
Comment (thatblindgeye) [inline packages/react-core/src/demos/PrimaryDetail.md:1125]: Updating the `activeCard` state to `null` when the drawer gets closed when clicking a selected card again, to remove the card styling/uncheck the hidden radio button.

File: packages/react-core/src/demos/PrimaryDetail.md [hunk 3]
```diff
@@ -1140,6 +1150,132 @@ class PrimaryDetailCardView extends React.Component {
...
+        return;
+      }
+
+      if ([13, 32].includes(event.keyCode)) {
+        if (event.currentTarget.id === this.state.activeCard) {
+          this.setState({
+            isDrawerExpanded: !this.state.isDrawerExpanded
+          });
+          return;
+        }
```
Comment (thatblindgeye) [inline packages/react-core/src/demos/PrimaryDetail.md:1270]: Same thing here

**Review by thatblindgeye — CHANGES_REQUESTED** [review]
The latest updates work great @jenny-s51! Just had one minor addition noted below based on these updates.

**Review by jenny-s51 — COMMENTED** [review]
Thank you @thatblindgeye ! Updated 👍

**Review by thatblindgeye — APPROVED** [review]
🎉 Awesome job on this demo!

**Pull request merged**